firmware/ventilator-controller-stm32/Core/Inc/Pufferfish/Driver/SPI/BMP388.h

```diff
@@ -0,0 +1,212 @@
+/*
+ * BMP388.h
+ * 
+ * Created on: Aug 29, 2020
+ *      Author: pradip.jm
+ */
+
+#pragma once
+
+#include "Pufferfish/HAL/STM32/HALSPIDevice.h"
+
+namespace Pufferfish {
+namespace Driver {
+namespace BMP388 {
+
+class SPIBMP388 {
+public:
+  SPIBMP388 (HAL::HALSPIDevice &spi):mSpi(spi){
```

Comment:
Drivers should never take as parameters any HAL modules which depend on the STM32 HAL; instead, they should take the respective abstract interfaces.
```suggestion
  SPIBMP388 (HAL::SPIDevice &spi):mSpi(spi){
```

Reply:
updated HAL::HALSPIDevice to HAL::SPIDevice